src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicAnalysisResourceTests.cs

```diff
@@ -244,6 +244,186 @@ public static void Main()
+            VerifySpans(reader, reader.Methods[1],
+                "(14,4)-(16,5)");
+
+            VerifySpans(reader, reader.Methods[2],
+                "(18,4)-(21,5)",
+                "(20,8)-(20,15)");
+
+            VerifySpans(reader, reader.Methods[3],
+                "(24,4)-(26,5)");
+
+            VerifySpans(reader, reader.Methods[4],
+                "(31,8)-(31,26)",
+                "(31,14)-(31,24)");
+
+            VerifySpans(reader, reader.Methods[5],
+                "(35,4)-(35,20)",
+                "(35,17)-(35,19)");
+
+            VerifySpans(reader, reader.Methods[6],
+                "(38,4)-(41,5)",
+                "(40,8)-(40,17)");
+
+            VerifySpans(reader, reader.Methods[7],
+                "(44,4)-(47,5)",
+                "(46,8)-(46,21)");
+
+            VerifySpans(reader, reader.Methods[8],
+                "(50,4)-(52,5)");
+
+            VerifySpans(reader, reader.Methods[9],
+                "(55,4)-(55,28)",
+                "(55,25)-(55,27)");
+
+            VerifySpans(reader, reader.Methods[10],
+                "(58,4)-(61,5)",
+                "(60,8)-(60,17)");
+        }
+
+        [Fact]
+        public void TestPatternSpans()
+        {
+            string source = @"
+using System;
+
+public class C
+{
+    public static void Main()                                   // Method 0
+    {
+        Student s = new Student();
+        s.Name = ""Bozo"";
+        s.GPA = 2.3;
+        Operate(s);
+    }
+     
+    static string Operate(Person p)                             // Method 1
+    {
+        switch (p)
+        {
+            case Student s when s.GPA > 3.5:
+                return $""Student {s.Name} ({s.GPA:N1})"";
+            case Student s:
+                return $""Student {s.Name} ({s.GPA:N1})"";
+            case Teacher t:
+                return $""Teacher {t.Name} of {t.Subject}"";
+            default:
+                return $""Person {p.Name}"";
+        }
+    }
+}
+
+class Person { public string Name; }
+class Teacher : Person { public string Subject; }
+class Student : Person { public double GPA; }
+";
+
+            var c = CreateCompilationWithMscorlib(Parse(source + InstrumentationHelperSource, @"C:\myproject\doc1.cs"));
+            var peImage = c.EmitToArray(EmitOptions.Default.WithInstrument("Test.Flag"));
+
+            var peReader = new PEReader(peImage);
+            var reader = DynamicAnalysisDataReader.TryCreateFromPE(peReader, "<DynamicAnalysisData>");
+
+            VerifySpans(reader, reader.Methods[0],
+                "(5,4)-(11,5)",
+                "(7,8)-(7,34)",
+                "(8,8)-(8,24)",
+                "(9,8)-(9,20)",
+                "(10,8)-(10,19)");
+
+            VerifySpans(reader, reader.Methods[1],
+                "(13,4)-(26,5)",
+                "(18,16)-(18,56)",
+                "(20,16)-(20,56)",
+                "(22,16)-(22,58)",
+                "(24,16)-(24,42)",
+                "(15,16)-(15,17)");
+        }
+
```

Comment:
@tmat Is there a test you could point to to illustrate how to verify sequence points?